src/Type.cc

```diff
@@ -992,39 +992,102 @@ void TypeDecl::DescribeReST(ODesc* d, bool roles_only) const
 		}
 	}
 
-// The following tracks how to initialize a given field, for fast execution
-// of Create().
+class DirectFieldInit final : public FieldInit
```

Comment:
Can you add some comments about what all of these individual init types do? It's kind of hard to track.

Reply:
Sounds good.